### ISSUE TITLE
Healthcheck script

### ISSUE THREAD

**Issue by kaznovac** [docker-library/memcached#91]
As a developer I'd like to have a way to issue a health check on memcached container without modifying the base image (base image has no `telnet` nor `nc`)

**Comment by LaurentGoderre** [comment]
The following works on alpine:

```
echo "version" | nc -vn -w 1 127.0.0.1 11211
```

Source: https://book.hacktricks.xyz/network-services-pentesting/11211-memcache

**Comment by kaznovac** [comment]
@LaurentGoderre Wow, thanks a lot, just what I've needed! (couldn't figure out the 'bidirectional' `cat` on my own)

I've added timeout to your debian command, so it will not block (just need to figure out the clean way to get the exit code)
```bash
echo "version" | (exec 3<>/dev/tcp/localhost/11211; cat >&3; timeout 0.1 cat <&3; exec 3<&-)
```

If it's ok with you I'd like to create PR to update the documentation of the image (later when I get some free time)

**Comment by tianon** [comment]
See https://github.com/docker-library/faq#healthcheck :see_no_evil:

**Comment by LaurentGoderre** [comment]
@tianon an example could be created here then: https://github.com/docker-library/healthcheck

**Comment by tianon** [comment]
Agreed!  Does `memcached` have an explicit healthcheck type function we could use, or is just connecting the best we can do?  For example, in SQL-based images, I've often used `SELECT 1` as a basic health check because I can verify that the output of that query is exactly `1` to know that everything actually worked end-to-end (and it wasn't just something else accepting my connection and quietly doing nothing useful).

**Comment by ghnp5** [comment]
Note - This only works if TCP port is enabled. Doesn't apply if we're using a sock file.
